utils/config.go

```diff
@@ -274,7 +274,13 @@ func newViper(allowEnvironmentOverrides bool) *viper.Viper {
 	defaults := getDefaultsFromStruct(model.Config{})
 
 	for key, value := range defaults {
-		v.SetDefault(key, value)
+		switch key {
```

Comment:
You might want to change this to just use an if statement to look for those keys. I know this works in Go since it doesn't automatically fall through switch cases, but for someone not experienced in the language, it looks like the switch statement does nothing